src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/Bau/BauCacheController.cs

```diff
@@ -0,0 +1,116 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using static GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces.IBlobStorageService;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Controllers.Api.Bau
+{
+    [Route("api/bau")]
+    [ApiController]
+    [Authorize]
+    public class BauCacheController : ControllerBase
+    {
+        private static readonly HashSet<string> AllowedPublicReleasePaths = new HashSet<string>
+        {
+            FileStoragePathUtils.PublicContentDataBlocksDirectory,
+            // TODO: EES-2865 Remove with other fast track code
+            FileStoragePathUtils.PublicContentFastTrackResultsDirectory,
+            FileStoragePathUtils.PublicContentSubjectMetaDirectory
+        };
+
+        private static readonly HashSet<string> AllowedPublicTreePaths = new HashSet<string>
+        {
+            PublicationTreeCacheKey.GetKey(PublicationTreeFilter.AnyData),
+            PublicationTreeCacheKey.GetKey(PublicationTreeFilter.LatestData),
+            PublicationTreeCacheKey.GetKey(null)
+        };
+
+        private readonly IBlobStorageService _privateBlobStorageService;
+        private readonly IBlobStorageService _publicBlobStorageService;
+
+        public BauCacheController(
+            IBlobStorageService privateBlobStorageService,
+            IBlobStorageService publicBlobStorageService)
+        {
+            _privateBlobStorageService = privateBlobStorageService;
+            _publicBlobStorageService = publicBlobStorageService;
+        }
+
+        [HttpDelete("private-cache")]
+        [Authorize(Roles = RoleNames.BauUser)]
+        public async Task<ActionResult> ClearPrivateCache()
+        {
+            // Remove all cache entries as we currently don't really need
+            // any more granularity currently (we only cache data blocks).
+            await _privateBlobStorageService.DeleteBlobs(BlobContainers.PrivateContent);
+
+            return NoContent();
+        }
+
+        [HttpDelete("public-cache/trees")]
+        [Authorize(Roles = RoleNames.BauUser)]
+        public async Task<ActionResult> ClearPublicCacheTrees(ClearCachePathsViewModel request)
+        {
+            var paths = request.Paths
+                .Where(AllowedPublicTreePaths.Contains)
+                .ToHashSet();
+
+            if (paths.Any())
+            {
+                await paths
+                    .ToAsyncEnumerable()
+                    .ForEachAwaitAsync(
+                        path =>
+                            _publicBlobStorageService.DeleteBlob(BlobContainers.PublicContent, path)
+                    );
+            }
+
+            return NoContent();
+        }
+
+        [HttpDelete("public-cache/releases")]
+        [Authorize(Roles = RoleNames.BauUser)]
+        public async Task<ActionResult> ClearPublicCacheReleases(ClearCachePathsViewModel request)
```

Comment:
Not suggesting a change here, but I think I would have been inclined to create more endpoints - one for data blocks, fast-tracks, etc. - rather than having the user provide the provides the directory names. It's a minor point, but my thought is that if someone unfamiliar to these endpoints comes to use them, their intention will be to just "remove all cached data blocks", and they could simply ping the right endpoint. 

As it stands, if you want to deleted all cached data blocks, it's not starkly obvious which endpoint you want. They will have to dig more deeply into this controller to understand which endpoint they want and what string to provide to delete data blocks.

That said, I accept that if you wanted to delete lots of different things - tree files / data blocks / etc. - and you already know how these endpoints work, you're potentially pinging quite a few endpoints if it was done the way I'm suggesting.

Reply:
Decided against changing it, but to improve the UX around this endpoint, I've spent some time adding a new `ContainsOnly` attribute and added `MinLength(1)`. This should make it more obvious what values are allowed in the `paths` field.

The `ContainsOnly` attribute should be fairly useful for other use-cases as well hopefully.